Part4/README.md

```diff
@@ -0,0 +1,182 @@
+# Creating a basic web backend API Part 4
+
+## Overview
+
+As our code base gets bigger, we need a better way to handle errors. For example, now if a request sent does not need the requirement, you use `res.send` to send a message. However, doing that in every single function is time intensive and a alot of code duplication!
+
+For example, if some one says to change the error message tomorrow to a JSON format instead, you will be plucking your hairs changing every line.
+
+So before we expand our code further, we will try to handle errors in a more elegant way.
+
+### Side track
+
+If you try an undefined route like `/whotttt` in the code from part3, you will get "NOT AUTHORIZED". Why this is happening? Because the code is read like a script and it as passed all defined routes and non match. When it hits the "decodeMiddleware" middleware, it runs a check on the body since it is empty, it will respond as "NOT AUTHORIZED".
```

Comment:
```suggestion
If you try an undefined route like `/whotttt` in the code from part3, you will get "NOT AUTHORIZED". Why is this happening? Because the code is read like a script and it has passed all defined routes and none match. When it hits the "decodeMiddleware" middleware, it runs a check on the body since it is empty, it will respond as "NOT AUTHORIZED".
```